src/csharp/Grpc.Core/Internal/AsyncCall.cs

```diff
@@ -407,12 +407,15 @@ private void HandleReceivedResponseHeaders(bool success, Metadata responseHeader
         /// <summary>
         /// Handler for unary response completion.
         /// </summary>
-        private void HandleUnaryResponse(bool success, ClientSideStatus receivedStatus, byte[] receivedMessage, Metadata responseHeaders)
+        private void HandleUnaryResponse(bool sucess, ClientSideStatus receivedStatus, byte[] receivedMessage, Metadata responseHeaders)
         {
+            // NOTE: because this event is a result of batch containing GRPC_OP_RECV_STATUS_ON_CLIENT,
+            // success will be always set to true.
```

Comment:
Are you sure about this? Does the batch *only* contain a `GRPC_OP_RECV_STATUS_ON_CLIENT` op?